web/controllers/group_controller.ex

```diff
@@ -7,10 +7,20 @@ defmodule Pairmotron.GroupController do
   plug :load_and_authorize_resource, model: Group, only: [:edit, :update, :delete]
 
   def index(conn, _params) do
-    groups = Repo.all(Group)
+    groups = Group |> order_by(:name) |> Repo.all
+    current_user = conn.assigns.current_user |> Repo.preload([:groups, :group_membership_requests])
+    conn = conn |> Plug.Conn.assign(:current_user, current_user)
     render(conn, "index.html", groups: groups)
   end
 
+  def show(conn, %{"id" => id}) do
+    group = Repo.get!(Group, id) |> Repo.preload(:owner)
+    invite_changeset = GroupMembershipRequest.changeset(%GroupMembershipRequest{}, %{group_id: id})
+    current_user = conn.assigns.current_user |> Repo.preload([:groups, :group_membership_requests])
+    conn = conn |> Plug.Conn.assign(:current_user, current_user)
+    render(conn, "show.html", group: group, invite_changeset: invite_changeset)
+  end
+
```

Comment:
Why did you move the show to before the :new action? Just curious, the default ordering that phoenix spawns for you has :show between the :create and :edit, which I think kind of makes sense since it can only be used on a resource that occurs after the :create action has been used.